plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -166,7 +166,7 @@ abstract class MavenPublishBaseExtension(
    * artifactIds like `[artifactId]-jvm`.
    */
   @Incubating
-  fun coordinates(groupId: String, artifactId: String, version: String) {
+  fun coordinates(groupId: String, artifactId: String, version: Any) {
```

Comment:
Thanks for the PR. What do you think about having an overload without `version` instead? The property will already default to it if it's not set (line 33) and Gradle will eventually (probably in v9) use a `Provider` for version so having `Any` here goes into the wrong direction when thinking about the future.

Reply:
The artifactId will default to the project name or to what the plugin that creates a publication uses. For example Kotlin multiplatform will use `{project.name}-{platform}`. So `artifactId` does not need to be called if the project name already has what you need.

Similar to the others `groupId` will default to `project.getGroup()`

Reply:
Ok, in that case, let me just drop the `version`-related commit, and limit this PR to the docs fix.

Reply:
Oh, I just realized that `groupId()` (as well as `version()` and `artifactId()`) are private, so I cannot call them directly. From that perspective, it could again make sense to have a `coordinates(group, artifactId)` overload. But then you'd probably want a `coordinates(group)` overload, too. Which leaves us with the question whether it's not better to simply make the underlying functions public instead of having `coordinates()` plus overloads.

However, changing the signature to

    fun coordinates(groupId: String? = null, artifactId: String? = null, version: String? = null)

could make sense, so users set only the needed parts in a single call (with named parameters).

Reply:
Like this: https://github.com/vanniktech/gradle-maven-publish-plugin/pull/691